src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -2420,129 +2523,159 @@ subroutine atm_advance_acoustic_step_work(nCells, nEdges, nCellsSolve, cellStart
 
         end do ! end loop over edges
+        !$acc end parallel
 
       end if ! test for first acoustic step
 
       if (small_step == 1) then  ! initialize here on first small timestep.
+         !$acc parallel
+         !$acc loop collapse(2)
          do iCell=cellStart,cellEnd
-            rtheta_pp_old(1:nVertLevels,iCell) = 0.0
+            do k=1,nVertLevels
+               rtheta_pp_old(k,iCell) = 0.0
+            end do
          end do
+         !$acc loop gang
+         do iCell=cellSolveStart,cellSolveEnd
+            !$acc loop vector
+            do k=1,nVertLevels
+               rho_pp(k,iCell) = 0.0
+               rtheta_pp(k,iCell) = 0.0
+               wwAvg(k,iCell) = 0.0
+               rw_p(k,iCell) = 0.0
+            end do
+            wwAvg(nVertLevels+1,iCell) = 0.0
+            rw_p(nVertLevels+1,iCell) = 0.0
+         end do
+         !$acc end parallel
       else
+         !$acc parallel
+         !$acc loop collapse(2)
          do iCell=cellStart,cellEnd
-            rtheta_pp_old(1:nVertLevels,iCell) = rtheta_pp(1:nVertLevels,iCell)
+            do k=1,nVertLevels
+               rtheta_pp_old(k,iCell) = rtheta_pp(k,iCell)
+            end do
          end do
+         !$acc end parallel
       end if
 
 !$OMP BARRIER
 
+      !$acc parallel
+      !$acc loop gang private(ts,rs)
       do iCell=cellSolveStart,cellSolveEnd  ! loop over all owned cells to solve
 
-         if(small_step == 1) then  ! initialize here on first small timestep.
-            wwAvg(1:nVertLevels+1,iCell) = 0.0            
-            rho_pp(1:nVertLevels,iCell) = 0.0            
-            rtheta_pp(1:nVertLevels,iCell) = 0.0            
-            rw_p(:,iCell) = 0.0
-         end if
-            
          if(specZoneMaskCell(iCell) == 0.0) then  ! not specified zone, compute...
 
-         ts(:) = 0.0
-         rs(:) = 0.0
+            !$acc loop vector
+            do k=1,nVertLevels
+               ts(k) = 0.0
+               rs(k) = 0.0
+            end do
 
-         do i=1,nEdgesOnCell(iCell) 
-            iEdge = edgesOnCell(i,iCell)
-            cell1 = cellsOnEdge(1,iEdge)
-            cell2 = cellsOnEdge(2,iEdge)
+            !$acc loop seq
+            do i=1,nEdgesOnCell(iCell)
+               iEdge = edgesOnCell(i,iCell)
+               cell1 = cellsOnEdge(1,iEdge)
+               cell2 = cellsOnEdge(2,iEdge)
 !DIR$ IVDEP
-            do k=1,nVertLevels
-               flux = edgesOnCell_sign(i,iCell)*dts*dvEdge(iEdge)*ru_p(k,iEdge) * invAreaCell(iCell)
-               rs(k) = rs(k)-flux
-               ts(k) = ts(k)-flux*0.5*(theta_m(k,cell2)+theta_m(k,cell1))
+               !$acc loop vector
+               do k=1,nVertLevels
+                  flux = edgesOnCell_sign(i,iCell)*dts*dvEdge(iEdge)*ru_p(k,iEdge) * invAreaCell(iCell)
+                  rs(k) = rs(k)-flux
+                  ts(k) = ts(k)-flux*0.5*(theta_m(k,cell2)+theta_m(k,cell1))
+               end do
             end do
```

Comment:
Question for @mgduda and @gdicker1 : We could perhaps rewrite this loop with an `acc reduction(+: flux.. etc)` instead of `acc loop seq`, but would our priority right now be to reorder the code as little as possible? I also have similar loops in the PR I'm working on. 

(I did quickly try using reduction for this loop, but it didn't really result in any performance improvement. But perhaps there might be something off in my code)

Reply:
I think there's value in trying to modify the code as little as possible during this phase of the porting work.

Reply:
I agree with both points - `ts` and `rs` seems like good candidates for a reduction clause here and we also shouldn't re-write the loop (yet). I'll keep this in mind for our optimization phase.

Reply:
Yeah, that makes sense. I'll also follow a similar approach for now with my PRs.